Famine - an ASR Ship Made for Holding Prisoners

Rubeebeebee opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
The Famine is a ship made for the ashen republic, able to hold prisoners and keep them safe even when under fire from other ships. It's also comes with facilities to "enhance" the quality of life of the prisoners, such as a botany bay, a kitchen, a medbay, chemistry bay, and a salvage bay, to both make sure the prisoners are alive and healthy, and to provide "alternative" entertainment solutions (AKA forcing the prisoners to farm stuff for your chemists to make fent with or something.)



## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->

For combat; its equipped with frankly thick armor plating, with 5 layers of plastitanium in total at any given side, albiet at the rear, this is an exception, since the rear only sports 3 layers, and the engine bay is located in a precarious spot, a side effect of it's original design being a civilian superfreighter. 

It is extremely undergunned compared to it's size, only sporting 2x Cyrexa 220m cannons, 2x ADEX Hybrid missile pods, 2x Rubicon EMP launchers, 2x RAC-30 autocannons, and finally, 2x 20mm Autocannons. 

It also cannot turn every well, it's gyroscope systems are very underpowered, so it'll have to rely on it's armor to stay alive and absorb the hits for it.

Overall; it's a Utility/Support ship that offers an alternative playstyle for the ASR, other than just fragging and pirating; taking in hostages and acting as a support ship in any fleet it's attached to.  

## How to test
<!-- Describe the way it can be tested -->
Load up the .yml in an IDE or buy it from blackmarket shipyard 

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
<img width="547" height="871" alt="famine ship whole" src="https://github.com/user-attachments/assets/53967975-3418-4e33-b2c8-676cd7635887" />
<img width="525" height="892" alt="famine complete" src="https://github.com/user-attachments/assets/65baaa3f-a36d-41d7-b1aa-f554f2051ae0" />
<img width="991" height="493" alt="famine live profile 3" src="https://github.com/user-attachments/assets/cbe51a77-2307-4bd1-b42b-9f83e2416144" />
<img width="1056" height="743" alt="famine live profile 2" src="https://github.com/user-attachments/assets/281440e3-7b5e-4cc0-81ab-7149bec66ba6" />
<img width="1026" height="713" alt="famine live profile 1" src="https://github.com/user-attachments/assets/7c70b871-2e0c-442c-aec3-e30159c9d30b" />
<img width="991" height="485" alt="famine profile 3" src="https://github.com/user-attachments/assets/f523e716-99d4-4a2b-9fa3-361f740308d8" />
<img width="1062" height="632" alt="famine profile 2" src="https://github.com/user-attachments/assets/a9784ad9-9997-463c-8d80-68e1fdc3ad50" />
<img width="1050" height="745" alt="famine profile 1" src="https://github.com/user-attachments/assets/a50e10ac-0475-4d0a-a0c7-35af3aba6b19" />
<img width="467" height="496" alt="big shit" src="https://github.com/user-attachments/assets/24771bd0-cb6b-4c99-bafc-a07a17c6b6e0" />
<img width="469" height="480" alt="famine warpoint" src="https://github.com/user-attachments/assets/1121a72c-8f6d-4a6a-a2b0-7e0fa6cb90f5" />
<img width="317" height="300" alt="famine latejoin" src="https://github.com/user-attachments/assets/fb7cccf5-83e5-439f-9ccf-85e78653f24c" />
<img width="945" height="212" alt="famine appraisal" src="https://github.com/user-attachments/assets/bd9b0ba5-4929-421f-ab67-357c37e7d40d" />

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
- [X] I can confirm this PR contains no AI-generated content, and did not use any AI-generated content.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

:cl:
- add: Added the HMA Famine, a ASR ship made to hold prisoners.

Rubeebeebee committed:
Famine - an ASR ship made for holding prisoners

monolith8319 committed:
chore: automatically update REUSE headers

Rubeebeebee commented:
@OneZerooo0 hiii boss, may i ask for a review?

tonotom1 commented:
> @OneZerooo0 hiii boss, may i ask for a review?

Hi Ruby epic ship but I think the answer be no til you fix your testfails

I think the logs are whining about two issues to fix

```
2025-09-12T01:55:32.6640236Z   Failed CheckAllShuttleGrids [1 m 36 s]
2025-09-12T01:55:32.6641348Z   Error Message:
2025-09-12T01:55:32.6646474Z    Multiple failures or warnings in test:
2025-09-12T01:55:32.6647522Z   1) SERVER: 27.937s [ERRO] entity_deserializer: Encountered invalid EntityUid reference wile reading entity 2463, component: DeviceNetwork Exception: 
2025-09-12T01:55:32.6649218Z   2) SERVER: 29.381s [ERRO] system.network_configurator: Can't resolve "Content.Shared.DeviceNetwork.Components.NetworkConfiguratorComponent" on entity 0/n0D!
```
Should be fixable deleting lines 20481 20482 and 20483 in your ship yml

```
2025-09-12T01:55:32.7168790Z   Failed NoShipyardShipArbitrage [1 m 37 s]
```
Should be fixable by doing an appraisegrid and raising your price higher to correlate with it (I might be wrong and it might just be just a double dip from the above error idk try fixing the above first and committing the change for testing)

Hope that helps

Rubeebeebee committed:
hopefully this fixes the tests

Rubeebeebee committed:
Merge branch 'Famine-Branch' of https://github.com/Rubeebeebee/Monolith into Famine-Branch

Rubeebeebee commented:
> > @OneZerooo0 hiii boss, may i ask for a review?
> 
> Hi Ruby epic ship but I think the answer be no til you fix your testfails
> 
> I think the logs are whining about two issues to fix
> 
> ```
> 2025-09-12T01:55:32.6640236Z   Failed CheckAllShuttleGrids [1 m 36 s]
> 2025-09-12T01:55:32.6641348Z   Error Message:
> 2025-09-12T01:55:32.6646474Z    Multiple failures or warnings in test:
> 2025-09-12T01:55:32.6647522Z   1) SERVER: 27.937s [ERRO] entity_deserializer: Encountered invalid EntityUid reference wile reading entity 2463, component: DeviceNetwork Exception: 
> 2025-09-12T01:55:32.6649218Z   2) SERVER: 29.381s [ERRO] system.network_configurator: Can't resolve "Content.Shared.DeviceNetwork.Components.NetworkConfiguratorComponent" on entity 0/n0D!
> ```
> 
> Should be fixable deleting lines 20481 20482 and 20483 in your ship yml
> 
> ```
> 2025-09-12T01:55:32.7168790Z   Failed NoShipyardShipArbitrage [1 m 37 s]
> ```
> 
> Should be fixable by doing an appraisegrid and raising your price higher to correlate with it (I might be wrong and it might just be just a double dip from the above error idk try fixing the above first and committing the change for testing)
> 
> Hope that helps

righto! did what you asked, raised the price to 500,000 and deleted the lines you mentioned

tonotom1 commented:
> righto! did what you asked, raised the price to 500,000 and deleted the lines you mentioned

OK I thought "arbitrage" meant the price test but it didn't do shit to remove the error I thought it would when you raised the price. Feel free to set the price back to previous because my fault for sucking at reading these logs.

Looking more closely you only needed to apply the first fix but that first fix was done incorrectly because FOR SOME REASON THERE'S A MISMATCH IN WHAT YOU SEE AND WHAT I SEE

<img width="579" height="288" alt="image" src="https://github.com/user-attachments/assets/61cfb7a3-9ba8-4d09-a5a1-fb88b7876143" />



The last three highlighted lines in this part is the offending part that actually needs to go. I don't know why Github gaslit like that. Like what the heck look it's the same line numbers but clearly different

<img width="332" height="181" alt="image" src="https://github.com/user-attachments/assets/16543d14-b787-443d-b756-6c33307921a5" />


What you were led to deleting was incorrect due to this (sorry!), you can undo the gaslighting mistake by downloading this version

https://github.com/Monolith-Station/Monolith/blob/1507f803e5df79aa2c681b516378c45f7481478c/Resources/Maps/_Mono/Shuttles/BlackMarket/famine.yml

And then remove the actual offending lines from there

Rubeebeebee committed:
removed the invalids in the ship .yml

Rubeebeebee commented:
> > righto! did what you asked, raised the price to 500,000 and deleted the lines you mentioned
> 
> OK I thought "arbitrage" meant the price test but it didn't do shit to remove the error I thought it would when you raised the price. Feel free to set the price back to previous because my fault for sucking at reading these logs.
> 
> Looking more closely you only needed to apply the first fix but that first fix was done incorrectly because FOR SOME REASON THERE'S A MISMATCH IN WHAT YOU SEE AND WHAT I SEE
> <img alt="image" width="579" height="288" src="https://private-user-images.githubusercontent.com/173343158/491146145-61cfb7a3-9ba8-4d09-a5a1-fb88b7876143.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgyMDg0ODUsIm5iZiI6MTc1ODIwODE4NSwicGF0aCI6Ii8xNzMzNDMxNTgvNDkxMTQ2MTQ1LTYxY2ZiN2EzLTliYTgtNGQwOS1hNWExLWZiODhiNzg3NjE0My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxOFQxNTA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MmIxYTc2YjcyYjI5MTk5ZGRjNzQ0ODM2NzY0NDFhMDM1NDQzNDA3ZTY1YjMyM2Y0ZTViY2YxODMwMDA5OWNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.usCq8DMGenQo3bm38wdbwgrOOZhGYJV-jQGJMCBeFGg">
> 
> The last three highlighted lines in this part is the offending part that actually needs to go. I don't know why Github gaslit like that. Like what the heck look it's the same line numbers but clearly different
> <img alt="image" width="332" height="181" src="https://private-user-images.githubusercontent.com/173343158/491146531-16543d14-b787-443d-b756-6c33307921a5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgyMDg0ODUsIm5iZiI6MTc1ODIwODE4NSwicGF0aCI6Ii8xNzMzNDMxNTgvNDkxMTQ2NTMxLTE2NTQzZDE0LWI3ODctNDQzZC1iNzU2LTZjMzMzMDc5MjFhNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxOFQxNTA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Nzk5MTg3OTQwMDQ4ZTc5Yzk2MDE3Njg4MjRjNTQ3NDQ2Y2FiNTRlMGMyYjRiNjAyYWRhZDI1M2ViNGMyNjQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rw5PHMCN9V5Obfs_IZaQx-JE3l52h_LHW1yRooSSnL8">
> 
> What you were led to deleting was incorrect due to this (sorry!), you can undo the gaslighting mistake by downloading this version
> 
> https://github.com/Monolith-Station/Monolith/blob/1507f803e5df79aa2c681b516378c45f7481478c/Resources/Maps/_Mono/Shuttles/BlackMarket/famine.yml
> 
> And then remove the actual offending lines from there

here we go! deleted the lines you requested, hopefully that fixes it

Rubeebeebee commented:
it worked!! thank you @tonotom1 !!!

tonotom1 commented:
It did

OneZerooo0 commented:
<img width="421" height="80" alt="image" src="https://github.com/user-attachments/assets/57216983-2553-4c82-a374-51445579abb1" />

We don't need another Jupiter-sized ship.... thats probably not going to be used.
Yeah, maybe a service ship, but theres hideout and CO theft for that, plus its just generally a bland concept for Monolith. Dealing with prisoners creatively is one thing, but having a whole station-ship specifically for that purpose is... not great.

graycatsaderow commented:
am i gonna get my money back or

Rubeebeebee commented:
> <img alt="image" width="421" height="80" src="https://private-user-images.githubusercontent.com/172208291/491284990-57216983-2553-4c82-a374-51445579abb1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgyNDM1ODAsIm5iZiI6MTc1ODI0MzI4MCwicGF0aCI6Ii8xNzIyMDgyOTEvNDkxMjg0OTkwLTU3MjE2OTgzLTI1NTMtNGM4Mi1hMzc0LTUxNDQ1NTc5YWJiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxOVQwMDU0NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02ZWM3MDM5ZTkwYmE5YmYwYzEzMmYxNDEwYjRlYWI4YmRmZTFiNmNhYzhiMGNmNDBmOTA4MTE3NGQ1N2RiYWMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.dPDuJ-wBjKVLnitnvAmXRmS1RpaJemGvOtDDb8gR8-w">
> 
> We don't need another Jupiter-sized ship.... thats probably not going to be used. Yeah, maybe a service ship, but theres hideout and CO theft for that, plus its just generally a bland concept for Monolith. Dealing with prisoners creatively is one thing, but having a whole station-ship specifically for that purpose is... not great.

is there a way to remake this to make it more acceptable?

@keeps47 commisioned me on this so i would like to make this work

OneZerooo0 commented:
Commission =/= merge

Rubeebeebee commented:
> Commission =/= merge

yes, i understand that, but if i can remake the ship in any way to make it acceptable, is there a way? is it possible?

OneZerooo0 commented:
Not sure, but my main concerns are:
- It’s way too big, redundant, and just… not very well thought out
- If it is implemented it will bloat the rogue shipyard with another vessel - especially bad if the ship isn’t good or used
- Do rogues really even need this? I’d argue not since there could be more creative things done

OneZerooo0 commented:
> am i gonna get my money back or

At least pay Rubeebee some if not all, they’ve done the due process

sleepyyapril commented:
> am i gonna get my money back or

a bounty is for making the ship, which they did. dont pay unless merged next time

Avalon-Proto commented:
Here's an idea - turn this into a hidden POI
